benchmark/main.cpp

```diff
@@ -156,7 +156,11 @@ do {                                                                        \
                                                                             \
     for (i = 1, alarmed = 0, t.attach(alarm, 1.0); !alarmed; i++)           \
     {                                                                       \
-        if ((ret = (CODE)) != 0) {                                          \
+        ret = CODE;                                                         \
+        if (ret == MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE) {                   \
```

Comment:
Have you tested this with changing the AES code to return `MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE` on 256 bit key? I think this will cause the benchmark to stop instead of skipping the test.

In addition, in future Unavailable errors, this will probably not work when we don't loop and overwrite the ret in next call, but this is not part of this PR

Reply:
>Have you tested this with changing the AES code to return MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE on 256 bit key? I think this will cause the benchmark to stop instead of skipping the test.

I did test it and it seems to work.

>In addition, in future Unavailable errors, this will probably not work when we don't loop and overwrite the ret in next call, but this is not part of this PR

Not sure what you mean. Please bear in mind that this piece of code is inside a while(0) loop. The break statement exits that while(0) (which is in the macro and will always be there. I dont think it really matters whether the whole macro itself is used within a loop or not.

Reply:
what I mean is, for AES256, for example:
Let's say that AES256_CBC will return the `MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE `, so the set key will succeed, but the `mbedtls_aes_crypt_cbc` will return the error code. `ret` is now the error `MBEDTLS_ERR_AES_FEATURE_UNAVAILABLE ` , and we finish the loop in [line 435](https://github.com/ARMmbed/mbed-os-example-tls/pull/194/files#diff-11d19f91de60e62a1952ab3e8b07cd0cR435) causing the function `benchmark_aes_cbc()` to return an error code, changing the `exit_code` to `MBEDTLS_EXIT_FAILURE`. We will not exit the test, but I wouldn't consider this a failure.

As for other modules, which will have a future NOT SUPPORTED error, it's basically the same, the specific benchmark test function will return an error code, even if we don't really consider a FEATURE UNAVAILABLE error a real error code

Reply:
Ok, I see what you mean. I will fix that.